manager/controllers/app/fybrikmodule_controller.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package app
+
+import (
+	"context"
+	"fmt"
+
+	"encoding/json"
+
+	validate "fybrik.io/fybrik/pkg/taxonomy/validate"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	api "fybrik.io/fybrik/manager/apis/app/v1alpha1"
+	"fybrik.io/fybrik/manager/controllers/utils"
+)
+
+// FybrikModuleReconciler reconciles a FybrikModule object
+type FybrikModuleReconciler struct {
+	client.Client
+	Name   string
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+// Reconcile reconciles FybrikModule CRD
+// It receives FybrikModule CRD and selects the appropriate modules that will run
+// The outcome is a Plotter containing multiple Blueprints that run on different clusters
+func (r *FybrikModuleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithValues("FybrikModule", req.NamespacedName)
+	// obtain FybrikModule resource
+	moduleContext := &api.FybrikModule{}
+	if err := r.Get(ctx, req.NamespacedName, moduleContext); err != nil {
+		log.V(0).Info("The reconciled object was not found")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// If the object has a scheduled deletion time, update status and return
+	if !moduleContext.DeletionTimestamp.IsZero() {
+		// The object is being deleted
+		return ctrl.Result{}, nil
+	}
+
+	observedStatus := moduleContext.Status.DeepCopy()
+	moduleVersion := moduleContext.GetGeneration()
+
+	// check if module has been validated before or if validated module is outdated
+	if string(moduleContext.Status.ValidModule) == "" || observedStatus.ValidatedGeneration != moduleVersion {
+		// do validation on moduleContext
+		err := ValidateFybrikModule(moduleContext, "/tmp/taxonomy/fybrik_module.json")
```

Comment:
Should the taxonomy location perhaps be in an environment variable rather than hard coded, or at least be in a constant?

Reply:
I made it a const variable